src/CMakeLists.txt

```diff
@@ -3,4 +3,90 @@
 # Copyright 2023 LunarG, Inc.
 #
 # SPDX-License-Identifier: Apache-2.0
+
+add_library(VulkanCompilerConfiguration INTERFACE)
+add_library(Vulkan::CompilerConfiguration ALIAS VulkanCompilerConfiguration)
```

Comment:
Let me know what you think of the name. I just picked something.

Linking against this library will apply the following compiler options/definitions/etc.